### PR TITLE
feat: adding default date format pattern for table, timeseries, bar and combochart cards

### PR DESCRIPTION
**Pull request by herleraja**
Closes https://github.com/carbon-design-system/carbon-addons-iot-react/issues/3720
Related to https://github.ibm.com/wiotp/Maximo-Asset-Monitor/issues/5050

**Summary**

- adding default date format pattern for table, timeseries, bar and combochart cards. we exposed a optionally property `defaultDateFormatPattern` which has a default value `L HH:mm:ss` 

**Change List (commits, features, bugs, etc)**

- feat: adding default date format pattern for table, timeseriesbar and combochart cards

**Acceptance Test (how to verify the PR)**

- Before: The default date format is `L HH:mm`
image
Now: The default date format is `L HH:mm:ss` --> https://deploy-preview-3721--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-card-tablecard--with-multiple-actions
image

**Regression Test (how to make sure this PR doesn't break old functionality)**

- All the cards should work without any issues. 

**Things to look for during review**

- [ ] Make sure all references to `iot` or `bx` class prefix is using the prefix variable
- [ ] (React) All major areas have a `data-testid` attribute. New test ids should have test written to ensure they are not changed or removed.
- [ ] UI should be checked in RTL mode to ensure the proper handling of layout and text.
- [ ] All strings should be translatable.
- [ ] The code should pass a11y scans (The storybook a11y knob should show no violations). New components should have a11y test written.
- [ ] Unit test should be written and should have a coverage of 90% or higher in all areas.
- [ ] All components should be passing visual regression test. For new styles or components either a visual regression test should be written for all permutations or the base image updated.
- [ ] Changes or new components should either write new or update existing documentation.
- [ ] PR should link and close out an existing issue

### CODE REVIEW

File: packages/react/src/constants/CardPropTypes.js [hunk 1]
```diff
@@ -160,7 +160,7 @@ export const TableCardPropTypes = {
           href: PropTypes.string,
           target: PropTypes.string,
         }),
-        /** optional type. If "TIMESTAMP", it will format to 'L HH:mm' */
+        /** optional type. If "TIMESTAMP", it will format to 'L HH:mm:ss' */
         type: PropTypes.string,
       })
```
Comment (jessieyan) [inline packages/react/src/constants/CardPropTypes.js:165]: The `defaultDateFormatPattern` prop us not added to the `CardPropTypes` for BarChartCard, ComboChartCard. TableCard and TimeSeriesCard?

**Review by jessieyan — CHANGES_REQUESTED** [review]

**Review by vaffonso — APPROVED** [review]

File: packages/react/src/components/TableCard/TableCard.jsx [hunk 2]
```diff
@@ -70,6 +70,7 @@ const defaultProps = {
   },
   renderDateDropdownInPortal: true,
   withToolbarTooltips: false,
+  defaultDateFormatPattern: 'L HH:mm:ss',
```
Comment (davidicus) [inline packages/react/src/components/TableCard/TableCard.jsx:73]: same comment about defaults

File: packages/react/src/components/TableCard/tableCardUtils.jsx [hunk 3]
```diff
@@ -23,7 +23,10 @@ export const determinePrecisionAndValue = (precision = 0, value, locale) => {
  * @param {Array<Object>} columns - Array of TableCard columns
  * @return {array} array of columns with formatted links and updated variable values
  */
-export const createColumnsWithFormattedLinks = (columns) => {
+export const createColumnsWithFormattedLinks = (
+  columns,
+  defaultDateFormatPattern = 'L HH:mm:ss'
```
Comment (davidicus) [inline packages/react/src/components/TableCard/tableCardUtils.jsx:28]: same comment about defaults

File: packages/react/src/components/ComboChartCard/comboChartHelpers.js [hunk 4]
```diff
@@ -188,6 +188,7 @@ export const useChartOptions = (content) => {
       size,
       timeDataSourceId,
       tooltipDateFormatPattern,
+      defaultDateFormatPattern,
```
Comment (davidicus) [inline packages/react/src/components/ComboChartCard/comboChartHelpers.js:191]: we should set the default like we do in the tableCardUtils file but using the old format, 'L HH:mm'

File: packages/react/src/components/TimeSeriesCard/TimeSeriesCard.jsx [hunk 5]
```diff
@@ -201,6 +201,7 @@ const defaultProps = {
   domainRange: null,
   tooltipDateFormatPattern: 'L HH:mm:ss',
   tooltipShowTotals: true,
+  defaultDateFormatPattern: 'L HH:mm:ss',
```
Comment (davidicus) [inline packages/react/src/components/TimeSeriesCard/TimeSeriesCard.jsx:204]: Same comment about defaults

**Review by davidicus — CHANGES_REQUESTED** [review]
@herleraja By adding a default where none existed before we are causing a breaking change to our down stream consumers. This looks good to go but we just need to add the default props to be the old value. This will probably allow us to remove the snapshot changes and the changes to the unit test. I would also like to see a unit test where we provide this prop and see that the value then changes.

File: packages/react/src/components/TableCard/TableCard.jsx (hunk 2 above)
Comment (herleraja) [reply packages/react/src/components/TableCard/TableCard.jsx]: Done

File: packages/react/src/components/ComboChartCard/ComboChartCard.jsx [hunk 6]
```diff
@@ -43,6 +43,7 @@ const defaultProps = {
   showTimeInGMT: false,
   domainRange: null,
   tooltipDateFormatPattern: 'L HH:mm:ss',
+  defaultDateFormatPattern: 'L HH:mm:ss',
```
[the remark this answers is not in the document]
Comment (herleraja) [reply packages/react/src/components/ComboChartCard/ComboChartCard.jsx]: Done

File: packages/react/src/components/BarChartCard/BarChartCard.jsx [hunk 7]
```diff
@@ -61,6 +61,7 @@ const defaultProps = {
   showTimeInGMT: false,
   tooltipDateFormatPattern: 'L HH:mm:ss',
   values: null,
+  defaultDateFormatPattern: 'L HH:mm:ss',
```
[the remark this answers is not in the document]
Comment (herleraja) [reply packages/react/src/components/BarChartCard/BarChartCard.jsx]: Done

File: packages/react/src/components/ComboChartCard/comboChartHelpers.js (hunk 4 above)
Comment (herleraja) [reply packages/react/src/components/ComboChartCard/comboChartHelpers.js:191]: Done

File: packages/react/src/components/TableCard/tableCardUtils.jsx (hunk 3 above)
Comment (herleraja) [reply packages/react/src/components/TableCard/tableCardUtils.jsx]: Done

**Review by vaffonso — APPROVED** [review]

**Review by davidicus — APPROVED** [review]

**Pull request merged**